view/ViewableData.php

```diff
@@ -384,17 +337,32 @@ protected function objCacheName($fieldName, $arguments) {
 	 * @return mixed
 	 */
 	protected function objCacheGet($key) {
-		if(isset($this->objCache[$key])) return $this->objCache[$key];
+		if(isset($this->objCache[$key])) {
+			return $this->objCache[$key];
+		}
+		return null;
 	}
 
 	/**
 	 * Store a value in the field cache
 	 *
 	 * @param string $key Cache key
 	 * @param mixed $value
+	 * @return $this
 	 */
 	protected function objCacheSet($key, $value) {
 		$this->objCache[$key] = $value;
+		return $this;
+	}
+
+	/**
+	 * Clear object cache
+	 *
+	 * @return $this
+	 */
+	protected function objCacheClear() {
```

Comment:
The problem is we don't know what caches will be invalidated. E.g. if you modify URLSegment, then the Link, RelativeLink, etc caches are all invalid.

The use case is VERY limited, as rarely are objects modified during rendering.